src/main/java/com/google/sps/servlets/BuddyServlet.java

```diff
@@ -0,0 +1,56 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.Constants;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/buddy")
+public class BuddyServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    String currentUserId = userService.getCurrentUser().getUserId();
+    String otherUserId = request.getParameter(Constants.BUDDY_ADD_PARAM);
+
+    Entity currentUserEntity = getUserEntity(currentUserId);
+    Entity otherUserEntity = getUserEntity(otherUserId);
+
+    List<String> currentBuddiesList = (List<String>) currentUserEntity.getProperty(Constants.USER_BUDDIES_PARAM);
+    List<String> otherBuddiesList = (List<String>) otherUserEntity.getProperty(Constants.USER_BUDDIES_PARAM);
+
+    // If the users are already buddies, remove buddies.
+    if (currentBuddiesList.contains(otherUserId)) {
+      currentBuddiesList.remove(otherUserId);
+      otherBuddiesList.remove(currentUserId);
+    
+    // If the users are not yet buddies, add buddies.
```

Comment:
Move this comment inside the body of the else. i.e. like: Same for the if-block.

```
else {
  // Some comment.
  ...
}